### PR TITLE
Random user agent method and data

### LINKED ISSUE #7: Random user agent
Create a data file of user agents to feed a random user-agent function.
Use this to feed `HTTPSource` requests.

Covers #6

### PR DESCRIPTION
**Pull request by eddowh**
Fixes #7

### CODE REVIEW

**Comment by hrbonz** [comment]
* Move `get_useragentstrings.py` to a folder called `utils`.
* Is that data format the best for simplicity? How are you going to select random user agents from this data structure?
* Create a `data` folder to hold the user agent list, it will also be useful for whatever we decide in #2.
* Where's my helper library that will help me use a random user agent? Create a `openkongqi.utils` module and get me a `get_rnd_ua()` in there.

**Comment by hrbonz** [comment]
The fetching tool was fine, just keep the whole fetching thing but add a destination file for the data dump on the command line using [argparse](https://docs.python.org/2.7/library/argparse.html).

**Comment by eddowh** [comment]
I think most, if not all, of the problems have been addressed. Except for the messages for catching the HTTP and URL errors when fetching.

Project structure now contains `data` and `utils` folders.

File: openkongqi/conf.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
     'REDIS_DB': 1,
     'SOURCES': {},
     'FEEDS': {},
+    'UA_STRINGS_FILE': os.path.join(os.path.abspath('..'), 'data/user_agent_strings.json')
```
Comment (hrbonz) [inline openkongqi/conf.py]: Do no put an absolute path here please, keep it relative from the running path.

File: openkongqi/utils.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+from .conf import settings
+
+
+def get_random_ua():
+    """Get random user agent string from list read from data.
+
+    :returns: str - a user agent string chosen randomly from list
+    """
+    with open(settings['UA_STRINGS_FILE'], 'r') as f:
```
Comment (hrbonz) [inline openkongqi/utils.py]: I don't like 1 letter variables except for loops (`for i in mylist`), I usually call File Descriptors `fd` for clarity. It's double the amount of letters but they're right next to each other ;)
What about catching exceptions if the file doesn't exist or is not readable?

File: utils/get_user_agent_strings.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    bs_start_time = time()
+    soup = BeautifulSoup(ua_list_resp)
+    bs_stop_time = time()
+    logger.debug("Converted in %.4f seconds" % (bs_stop_time - bs_start_time))
+    liste = soup.find('div', {'id': 'liste'})
+    ua_strings = map(lambda x: x.text, liste.find_all('li'))
+    return ua_strings
+
+
+def setup_argparser():
```
Comment (hrbonz) [inline utils/get_user_agent_strings.py]: This function doesn't bring much, you could merge it in `main()`

File: utils/get_user_agent_strings.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    logger.debug("Set output file as {}"
+                 .format(os.path.abspath(args.file.name)))
+    # scrape html and extract content
+    ua_list = []
+    for category, url in UA_LINKS.items():
+        user_agents = extract_user_agents(
+            fetch_ua_url_content(url, category))
+        ua_list.extend(user_agents[:3])  # just the first three
+    # dump content into file as specified from command line
+    with open(os.path.abspath(args.file.name), 'w') as f:
```
Comment (hrbonz) [inline utils/get_user_agent_strings.py]: why do you get the absolute path here? Just open the file as it was passed to the command line, if there's a problem it's up to the person running the script to fix it.

File: openkongqi/conf.py (hunk 1 above)
Comment (hrbonz) [inline openkongqi/conf.py]: Just put the path string, settings should be as explicit as possible.

File: utils/get_user_agent_strings.py (hunk 4 above)
Comment (eddowh) [inline utils/get_user_agent_strings.py]: Should I change every `abspath` or can I keep the log displaying the absolute path? Is that considered back practice too?

Comment (hrbonz) [inline utils/get_user_agent_strings.py]: Yes, what I want to see in the logs is what I put in the settings. If I change folder for some reason, then the meaning is completely different and the absath is going to give me a different thing as well.

File: utils/get_user_agent_strings.py (hunk 3 above)
Comment (eddowh) [inline utils/get_user_agent_strings.py]: You mean both the `extract_user_agents` and the `setup_argparser` function?

**Comment by hrbonz** [comment]
In the settings module, check for the existence and readability of the UA file. raise a `ConfigError` if the file is not accessible properly.

**Comment by hrbonz** [comment]
To complete this, let's add the random call to `HTTPSource`. Remove the `self.ua` from `__init__` and call for a random ua in `HTTPSource.get_req()`

File: openkongqi/conf.py [hunk 5]
```diff
@@ -67,6 +68,17 @@ def get_sources():
...
+# check for existence and readability of UA_STRINGS_FILE;
+# catch ConfigError as early as possible
+try:
+    open(settings['UA_STRINGS_FILE'], 'r')
+except IOError as e:
+    err_reason = {
+        2: 'nonexistent',
+        13: 'unreadable'
+    }
+    raise ConfigError("file is {}".format(err_reason[e.errno])
```
Comment (hrbonz) [inline openkongqi/conf.py]: Simply relay the exception error message.

**Pull request merged**